include/mc_rtc/gui/Form.h

```diff
@@ -189,28 +189,43 @@ struct FormComboInput : public FormElement<FormComboInput, Elements::ComboInput>
   inline FormComboInput(const std::string & name,
                         bool required,
                         const std::vector<std::string> & values,
+                        int def,
                         bool send_index = false)
-  : FormElement<FormComboInput, Elements::ComboInput>(name, required), values_(values), send_index_(send_index)
+  : FormElement<FormComboInput, Elements::ComboInput>(name, required), values_(values), def_(def),
+    send_index_(send_index), has_def_(true)
   {
   }
 
+  inline FormComboInput(const std::string & name,
+                        bool required,
+                        const std::vector<std::string> & values,
+                        bool send_index = false)
+  : FormComboInput(name, required, values, -1, send_index)
+  {
+    has_def_ = false;
+  }
+
   static constexpr size_t write_size_()
   {
-    return 2;
+    return 4;
   }
 
   inline void write_(mc_rtc::MessagePackBuilder & builder)
   {
     builder.write(values_);
+    builder.write(def_);
     builder.write(send_index_);
+    builder.write(has_def_);
```

Comment:
For compatibility, the new fields should be written last otherwise old client will interpret `def_` as `send_index_`

Reply:
Thanks. Let me double check before starting revise. `has_def_` can be deleted here and also from all changes in https://github.com/jrl-umi3218/mc_rtc_ros/pull/24 ?

Reply:
Yes. I think it's reasonable to give an `int` to the client and have it handle `< 0` as "no-default given"